File: mq-spring/src/main/java/com/example/mq_spring/entity/Partner.java
package com.example.mq_spring.entity;

import jakarta.persistence.*;
import lombok.*;

@Getter
@Setter
@Builder
@AllArgsConstructor
@NoArgsConstructor
@Entity
public class Partner {
    @Id
    private @GeneratedValue Long id;

    private String alias;

    private String type;

    @Enumerated(EnumType.STRING)
    private DirectionEnum direction;

    private String application;

    @Enumerated(EnumType.STRING)
    private ProcessedFlowTypeEnum processedFlowType;

    private String description;

}

File: mq-angular/src/app/partner/partner.ts
export interface Partner {
  id: number;
  alias: string;
  type: string;
  direction: DirectionEnum;
  application: string;
  processedFlowType: ProcessedFlowTypeEnum;
  description: string
}

export enum DirectionEnum {
  INBOUND = 'INBOUND',
  OUTBOUND = 'OUTBOUND'
}

export enum ProcessedFlowTypeEnum {
  MESSAGE = 'MESSAGE',
  ALERTING = 'ALERTING',
  NOTIFICATION = 'NOTIFICATION'
}
